File: 125-ValidPalindrome/125-ValidPalindrome.cpp
// Last updated: 1/14/2026, 5:18:25 PM
#include <string>
#include <cctype>

using std::tolower;
using std::string;

class Solution {
public:
    bool isPalindrome(string s) {
        string fullStringNoSpace = "";
        for(unsigned int i = 0; i < s.size(); ++i) {
            if(isalpha(s[i]) || isdigit(s[i])) {
                fullStringNoSpace += tolower(s[i]);
            }
        }
        for(unsigned int i = 0; i < fullStringNoSpace.size() / 2; ++i) {
            if(fullStringNoSpace[i] != fullStringNoSpace[fullStringNoSpace.size() - 1 - i]) {
                return false;
            }
        }
        return true;
    }
};

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 1/14/2026, 5:18:22 PM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        newString = ""
        for i in s:
            if i.isalpha() or i.isdigit():
                newString += i.lower()
        print(newString)

        for i in range(len(newString)):
            if newString[i] != newString[len(newString) - i - 1]:
                return False
        return True
        